### PR TITLE
Feature/long grid name

### PR DESCRIPTION
**Pull request by jedwards4b**
allows the complete long grid name as an argument to create_newcase

Test suite:
Test baseline:
Test namelist changes:
Test status: bit for bit
Fixes 

User interface changes?:

Update gh-pages html (Y/N)?:

### CODE REVIEW

**Review by jasonb5 — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "CIME/non_py/cprnc"]
 	path = CIME/non_py/cprnc
-	url = git@github.com:ESMCI/cprnc
+	url = https://github.com/ESMCI/cprnc
```
[the remark this answers is not in the document]
Comment (gold2718) [reply .gitmodules:3]: Doesn't this suggested change require anyone trying to use CIME to have a GitHub account with registered SSH keys? That is not currently a requirement for CESM, is it for E3SM?

[the remark this answers is not in the document]
Comment (jedwards4b) [reply .gitmodules:3]: Yes, we have discussed this - e3sm requires the ssh protocol for jenkins and cesm is currently using a branch of cime using https, the git-fleximod will handle this correctly without a branch so creating this branch is a temporary requirement for cesm.

**Review by jasonb5 — APPROVED** [review]
LGTM

**Review by mvertens — APPROVED** [review]
Thanks Jim. I've tested this and it looks good. This will be super useful to have.

File: CIME/XML/grids.py [hunk 2]
```diff
@@ -94,6 +110,17 @@ def get_grid_info(self, name, compset, driver):
 
         return gridinfo
 
+    def _valid_lname(self, name):
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply CIME/XML/grids.py:113]: I ran SMS_Ln9.T42z30_T42_mg17.FHS94.derecho_intel and SMS_Ln9.a%T42z30_l%null_oi%null_r%null_g%null_w%null_z%null_m%gx1v7.2000_CAM%HS94_SLND_SICE_SOCN_SROF_SGLC_SWAV_SESP.derecho_intel  and compared results.  They are identical.

**Pull request merged**